disco/bot/bot.py

```diff
@@ -383,11 +411,17 @@ def handle_message(self, msg):
         bool
             Whether any commands where successfully triggered by the message.
         """
+        if not self.config.commands_require_mention and self.config.commands_prefix_getter:
+            custom_message_prefixes = self.config.commands_prefix_getter(msg)
+        else:
+            custom_message_prefixes = []
+
         commands = list(self.get_commands_for_message(
             self.config.commands_require_mention,
             self.config.commands_mention_rules,
             self.config.commands_prefix,
             msg,
+            self.config.commands_prefixes + custom_message_prefixes,
```

Comment:
I'd argue it makes more sense for `custom_message_prefixes` to not be merged with the original commands_prefixes. This gives the end user more control when using the functional version of this configuration option (e.g. blocking some prefixes from matching in certain channels, or having server specific prefixes).